actionpack/lib/action_dispatch/middleware/cookies.rb

```diff
@@ -489,11 +497,26 @@ def []=(name, options)
         def request; @parent_jar.request; end
 
       private
-        def expiry_options(options)
-          if options[:expires].respond_to?(:from_now)
-            { expires_in: options[:expires] }
+        def fetch_metadata(options)
+          { purpose: options[:cookie_name] }.tap do |metadata|
```

Comment:
Think the `tap` here blurs the flow. Is building the metadata at the same level of abstraction as normalizing the expiry? If not, how can we compose this code better?